File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/LeaveBalanceColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LeaveBalanceRow } from "./LeaveBalanceRow";
import { LeaveType } from "./LeaveType";

export interface LeaveBalanceColumns {
    EmployeeFullName: Column<LeaveBalanceRow>;
    LeaveType: Column<LeaveBalanceRow>;
    EmployeePaidLeavesPerMonth: Column<LeaveBalanceRow>;
    EmployeeJoinDate: Column<LeaveBalanceRow>;
    MonthsWorked: Column<LeaveBalanceRow>;
    AccruedLeaves: Column<LeaveBalanceRow>;
    LeavesThisMonth: Column<LeaveBalanceRow>;
    LeavesPreviousMonths: Column<LeaveBalanceRow>;
}

export class LeaveBalanceColumns extends ColumnsBase<LeaveBalanceRow> {
    static readonly columnsKey = 'Operations.LeaveBalance';
    static readonly Fields = fieldsProxy<LeaveBalanceColumns>();
}

[LeaveType]; // referenced types

File: HRMS/HRMS.Web/Modules/Operations/LeaveBalance/LeaveBalanceColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.LeaveBalance")]
[BasedOnRow(typeof(LeaveBalanceRow), CheckNames = true)]
public class LeaveBalanceColumns
{
    [EditLink, Width(200)]
    public string EmployeeFullName { get; set; }
    public LeaveType LeaveType { get; set; }
    public int EmployeePaidLeavesPerMonth { get; set; }
    public DateTime EmployeeJoinDate { get; set; }
    public int MonthsWorked { get; set; }
    public int AccruedLeaves { get; set; }
    public decimal LeavesThisMonth { get; set; }
    public decimal LeavesPreviousMonths { get; set; }
}
